worlds/undertale/Rules.py

```diff
@@ -101,7 +101,10 @@ def set_rules(multiworld: MultiWorld, player: int):
         set_rule(multiworld.get_location("True Lab Plot", player),
                  lambda state: state.can_reach("New Home", "Region", player)
                                and state.can_reach("Letter Quest", "Location", player)
-                               and state.can_reach("Alphys Date", "Location", player))
+                               and state.can_reach("Alphys Date", "Location", player)
+                               and (state.has("Left Home Key", player) and
+                                    state.has("Right Home Key", player)) or
```

Comment:
Fairly certain this paren is supposed to go after the next line
```suggestion
                                    state.has("Right Home Key", player) or
```